File: src/fib/fib.js
function fib(n) {
    if(n == 0)
        return 0
    else if(n > 0 && n <= 2)
        return 1
    else
        return fib(n - 1) + fib(n - 2)
}

async function a_fib(n) {
    if(n == 0)
        return 0
    else if(n > 0 && n <= 2)
        return 1
    else
        resp1 = a_fib(n - 1)
        resp2 = a_fib(n - 2)
        return await resp1 + await resp2
}


t = new Date()
result = fib(40)
tf = new Date() - t
console.log(`Total time elapsed: ${tf/1000} seconds; Result: ${result}`)

// t = new Date()
// result = a_fib(40)
// tf = new Date() - t
// console.log(`Total time elapsed: ${tf/1000} seconds; Result: ${result}`)

File: src/fib/fib.py
import time
import asyncio

def fib(n):
    if n == 0:
        return 0
    elif 0 < n <= 2:
        return 1
    else:
        return fib(n - 1) + fib(n - 2)

async def a_fib(n):
    if n == 0:
        return 0
    elif 0 < n <= 2:
        return 1
    else:
        return sum(await asyncio.gather(a_fib(n - 1), a_fib(n - 2)))

t = time.perf_counter()
result = fib(40)
tf = time.perf_counter() - t
print(f"Total time elapsed: {tf} seconds; Result: {result}")

# t = time.perf_counter()
# result = asyncio.run(a_fib(40))
# tf = time.perf_counter() - t
# print(f"Total time elapsed: {tf} seconds; Result: {result}")
